sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -205,8 +230,45 @@ export function convertResponseForMetricsDefinitions(
  */
 export function convertResponseForMetricNamespaces(
   generatedResponse: GeneratedMetricNamespacesListResponse
-): GetMetricNamespacesResponse {
+): GetMetricNamespacesResult {
   return {
     namespaces: generatedResponse.value
   };
 }
+
+/**
+ * @internal
+ */
+export function convertGeneratedTable(table: GeneratedTable): LogsTable {
+  const dynamicsIndices: number[] = [];
+  const datesIndices: number[] = [];
+
+  // most columns convert on deserialization except for `dynamic` columns (basically JSON objects)
+  // and 'datetime' (strings that are ISO8601 formatted dates)
+  for (let i = 0; i < table.columns.length; ++i) {
+    if (table.columns[i].type === "datetime") {
+      datesIndices.push(i);
+    } else if (table.columns[i].type === "dynamic") {
+      dynamicsIndices.push(i);
+    }
+  }
+
+  return {
+    ...table,
+    rows: (table.rows as LogsTable["rows"]).map((row) => {
+      for (const dynamicIndex of dynamicsIndices) {
+        try {
+          row[dynamicIndex] = JSON.parse(row[dynamicIndex] as string) as Record<string, unknown>;
+        } catch (_err) {
+          /* leave as is. */
```

Comment:
Wondering if we should log this - not sure though...

Reply:
This is an interesting one and I'm not sure yet either. I need to toy around more with the API so I was just playing it safe since failing to deserialize a single value shouldn't tank the entire query result.